src/validator.ts

```diff
@@ -11,106 +11,100 @@ import {
   VariableDeclaration,
 } from 'solc-typed-ast';
 
-export function validate(node: NodeToProcess, natspec: Natspec, config: Config): string[] {
-  // There is inheritdoc, no other validation is needed
-  if (natspec.inheritdoc) return [];
+export class Validator {
+  config: Config;
 
-  // Inheritdoc is enforced but not present, returning an error
-  if (config.enforceInheritdoc && requiresInheritdoc(node)) return [`@inheritdoc is missing`];
-
-  // Validate natspec for the constructor only if configured
-  if (node instanceof FunctionDefinition && node.kind === 'constructor') {
-    return config.constructorNatspec ? validateParameters(node, natspec) : [];
+  constructor(config: Config) {
+    this.config = config;
   }
 
-  // Inheritdoc is not enforced nor present, and there is no other documentation, returning error
-  if (!natspec.tags.length) return [`Natspec is missing`];
-
-  // Validate the completeness of the documentation
-  let alerts: string[] = [];
-  if (node instanceof EnumDefinition) {
-    // TODO: Process enums
-  } else if (node instanceof ErrorDefinition) {
-    alerts = [...alerts, ...validateParameters(node, natspec)];
-  } else if (node instanceof EventDefinition) {
-    alerts = [...alerts, ...validateParameters(node, natspec)];
-  } else if (node instanceof FunctionDefinition) {
-    alerts = [...alerts, ...validateParameters(node, natspec), ...validateReturnParameters(node, natspec)];
-  } else if (node instanceof ModifierDefinition) {
-    alerts = [...alerts, ...validateParameters(node, natspec)];
-  } else if (node instanceof StructDefinition) {
-    alerts = [...alerts, ...validateMembers(node, natspec)];
-  } else if (node instanceof VariableDeclaration) {
-    // Only the presence of a notice is validated
-  }
+  validate(node: NodeToProcess, natspec: Natspec): string[] {
```

Comment:
shouln't all methods but this one be private? or is it for testing purposes?

Reply:
> or is it for testing purposes?
Initially yes, but now I see a way to test the `validate` function fully without exposing the internal parts.